invenio_rdm_records/services/services.py

```diff
@@ -128,3 +128,19 @@ def oai_result_item(self, identity, oai_record_source):
             record,
             links_tpl=self.links_item_tpl,
         )
+
+    @unit_of_work()
+    def reindex_user_records(self, identity, id_, uow=None):
+        """Reindexes all the records from a given user."""
+        self.require_permission(identity, "manage")
```

Comment:
Not 100% sure about this permission.

The service exposes one "reindex" method in which permissions are set to `search`. 

However, for 'access' we usually require `manage` permissions